pulsar-client-cpp/lib/Schema.cc

```diff
@@ -90,6 +131,43 @@ SchemaInfo::SchemaInfo(SchemaType schemaType, const std::string &name, const std
                        const StringMap &properties)
     : impl_(std::make_shared<SchemaInfoImpl>(schemaType, name, schema, properties)) {}
 
+SchemaInfo::SchemaInfo(const SchemaInfo &keySchema, const SchemaInfo &valueSchema,
+                       const KeyValueEncodingType &keyValueEncodingType) {
+    std::string keySchemaStr = keySchema.getSchema();
+    std::string valueSchemaStr = valueSchema.getSchema();
+    int keySize = keySchemaStr.size();
+    int valueSize = valueSchemaStr.size();
+
+    int buffSize = sizeof keySize + keySize + sizeof valueSize + valueSize;
```

Comment:
```suggestion
    size_t keySize = keySchemaStr.size();
    size_t valueSize = valueSchemaStr.size();

    size_t buffSize = sizeof keySize + keySize + sizeof valueSize + valueSize;
```

or use `auto`. In C++, conversion between unsigned and signed integers might cause some potential bugs. If the conversion cannot be avoided, we'd better use `static_cast` to mark the conversion explicitly.